pkg/logging/plain-handler.go

```diff
@@ -0,0 +1,51 @@
+package logging
+
+import (
+	"context"
+	"io"
+	"sync"
+	"unicode"
+
+	"golang.org/x/exp/slog"
+)
+
+type plainHandler struct {
+	mu    sync.Mutex
+	w     io.Writer
+	level slog.Level
+}
+
+func newPlainHandler(w io.Writer, level slog.Level) *plainHandler {
+	return &plainHandler{
+		mu:    sync.Mutex{},
+		w:     w,
+		level: level,
+	}
+}
+
+func (ph *plainHandler) Enabled(_ context.Context, level slog.Level) bool {
+	return level >= ph.level
+}
+
+func (ph *plainHandler) Handle(ctx context.Context, r slog.Record) error {
+	runes := []rune(r.Message)
+
+	// upper case the first letter
+	if len(runes) > 0 {
+		runes[0] = unicode.ToUpper(runes[0])
+	}
+	runes = append(runes, '\n')
+
+	ph.mu.Lock()
+	_, err := ph.w.Write([]byte(string(runes)))
+	defer ph.mu.Unlock()
```

Comment:
I can't find the rant post from Dave Cheney right now, but deferred unlocks are discouraged (unless _really_ needed of course) because:

1. A `defer` statement _always_ causes significant overhead, no matter how simple the called function is. Since you're at the end of the method anyways and nothing else happens, there's no reason to defer it.
2. Minimizing the time of lock: The only operation that needs locking here is the `Write()`, and unlocking takes place regardless of error. Nothing changes if you just remove the `defer` statement.

Reply:
Very good point about the `defer`. I guess the `defer` only makes sense if there's a lot of work happening that needs to be protected.